File: examples/javascript/load-prompts.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const repoRoot = path.resolve(__dirname, "..", "..");
const catalogPath = path.join(repoRoot, "data", "catalog.json");

const catalog = JSON.parse(fs.readFileSync(catalogPath, "utf8"));

function filterEntries(entries, { category, excludeStatuses = [] } = {}) {
  return entries.filter((entry) => {
    if (category && entry.category !== category) {
      return false;
    }
    if (excludeStatuses.includes(entry.status)) {
      return false;
    }
    return true;
  });
}

const curatedImagePrompts = filterEntries(catalog.entries, {
  category: "image",
  excludeStatuses: ["seed"],
});

for (const entry of curatedImagePrompts) {
  console.log(`${entry.id}: ${entry.title} [${entry.status}] -> ${entry.path}`);
}

File: examples/python/load_prompts.py
from __future__ import annotations

import json
from pathlib import Path


REPO_ROOT = Path(__file__).resolve().parents[2]
CATALOG_PATH = REPO_ROOT / "data" / "catalog.json"


def filter_entries(entries: list[dict], *, category: str | None = None, exclude_statuses: set[str] | None = None) -> list[dict]:
    excluded = exclude_statuses or set()
    return [
        entry for entry in entries
        if (category is None or entry["category"] == category)
        and entry["status"] not in excluded
    ]


def main() -> None:
    catalog = json.loads(CATALOG_PATH.read_text())
    curated = filter_entries(
        catalog["entries"],
        category="image",
        exclude_statuses={"seed"},
    )

    for entry in curated:
        print(f'{entry["id"]}: {entry["title"]} [{entry["status"]}] -> {entry["path"]}')


if __name__ == "__main__":
    main()
